rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/RxBleConnectionImpl.java

```diff
@@ -83,6 +83,21 @@ public RxBleConnectionImpl(
         this.longWriteOperationBuilderProvider = longWriteOperationBuilderProvider;
         this.callbackScheduler = callbackScheduler;
         this.illegalOperationChecker = illegalOperationChecker;
+
+        // unsubscription is not needed as all objects are on @ConnectionScope and will get GCed together
```

Comment:
The assumption on object lifecycle from within the object is generally not a good pattern, even if in this case it's true.

Reply:
Agreed